### DISCUSSION TITLE
DuckDB Settings Across Multiple Processes In Memory Clarification Needed

### DISCUSSION THREAD

**Discussion by brianwyka** [duckdb/duckdb#11155] [Q&A]
We are spawning multiple DuckDB processes on linux using the CLI to run in parallel.  We are using `memory` mode connection.

We are configuring each process for example with:

```sql
SET memory_limit = '8GB';
```

If we have `5` processes running in parallel, will DuckDB try to use `8GB` for all the processes in total, or will it try to consume `40GB`, `8GB` for each of the `5` processes?

In general, when configuring a setting on a standalone process, does that setting apply to that process alone, or to all?

**Comment by Tishj** [comment]
It applies to that process alone

**Reply by brianwyka** [reply]
Thanks @Tishj .  With respect to the other APIs, like Java where we can have multiple concurrent connections on an in-memory DB, I'm assuming that the setting will be the following:

Per connection settings, so if there are `5` concurrent connections, there will be `40GB` total in use:
```java
final Properties connectionProperties = new Properties();
connectionProperties.setProperty("memory_limit", "8GB");

try (final Connection connection = DriverManager.getConnection("jdbc:duckdb:memory:", connectionProperties)) {

}
```

Is that also correct?

Does it differ if the connection is duplicated?

```java
DuckDBConnection conn2 = (DuckDBConnection) connection).duplicate();
```

**Reply by Tishj** [reply]
I believe memory_limit is a DatabaseConfig option, meaning this setting is shared accross all connections.

And this has impact on the BufferManager, which is part of the database, not a connection

So setting this to 5gb will limit it to 5gb, no matter how many connections exist to the database

**Reply by Tishj** [reply]
Note that even with ATTACH this doesn't count as a new "database" in this context, and it will share the same buffer manager

**Reply by brianwyka** [reply]
Thanks @Tishj
